### ISSUE TITLE
Add ICaRL to the CL Baselines

### ISSUE THREAD

**Issue by vlomonaco** [ContinualAI/avalanche#4]
Add the ICaRL strategy to the main CL Baselines.

**Comment by vlomonaco** [comment]
This is gonna be added soon with a PR by Andrea Rosasco.

**Comment by Mathieu4141** [comment]
Are you still working on that @andrew-r96 ? For a school project we implemented iCaRL as a baseline, and we did it using avalanche. The code is not up to date with the latest version of avalanche, but it shouldn't be too much work to clean it.
Let me know if you're still on it :)

**Comment by andrew-r96** [comment]
Hi @Mathieu4141 thanks for the interest :)
Yeah I'm still working on it and  just opened a PR here https://github.com/ContinualAI/avalanche/pull/637 if you wanna take a look.

**Comment by Mathieu4141** [comment]
Great! Yes I'll take a look at your PR 🙂

**Comment by qsunyuan** [comment]
Great work.

Hi!

Great work.

Recently I also reproduced iCaRL, but I cannot achieve the original results. I tried many methods, adjust learning rate, more training epochs, different weight decay.....

Could u pls share ur CIFAR100 icarl results.

For example, ten tasks which each task has 10 classes.
 
I cannot achieve the original results, about 49%. How's ur results.

<img width="327" alt="1641972564(1)" src="https://user-images.githubusercontent.com/71776283/149083010-93d0fd24-d2bf-47dc-b8ec-4c779484cf49.png">

Hope to get ur replay. Best wishes.

**Comment by qsunyuan** [comment]
Hi!

Great work.

Recently I also reproduced iCaRL, but I cannot achieve the original results.

Could u pls share ur CIFAR100 icarl results.

For example, ten tasks which each task has 10 classes.

I cannot achieve the original results, about 49%. How's ur results.

<img width="326" alt="1641972671(1)" src="https://user-images.githubusercontent.com/71776283/149083213-5ad014d1-05f5-45a7-ba4d-467215823482.png">


Hope to get ur replay. Best wishes.

**Comment by vlomonaco** [comment]
Hi @qsunyuan! Results on Cifar100 are reproduced here: https://github.com/ContinualAI/reproducible-continual-learning

**Comment by qsunyuan** [comment]
THx for ur help, it helps me a lot. Best wishes.
